benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/RedisHubLifetimeManagerBenchmark.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.AspNetCore.SignalR.Redis;
+using Microsoft.AspNetCore.SignalR.Redis.Tests;
+using Microsoft.AspNetCore.SignalR.Tests;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.SignalR.Microbenchmarks
+{
+    public class RedisHubLifetimeManagerBenchmark
+    {
+        private RedisHubLifetimeManager<TestHub> _manager1;
+        private RedisHubLifetimeManager<TestHub> _manager2;
+        private TestClient[] _clients;
+        private object[] _args;
+        private List<string> _excludedIds = new List<string>();
+        private List<string> _sendIds = new List<string>();
+        private List<string> _groups = new List<string>();
+        private List<string> _users = new List<string>();
+
+        private const int ClientCount = 20;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            var server = new TestRedisServer();
+            var logger = NullLogger<RedisHubLifetimeManager<TestHub>>.Instance;
+            var options = Options.Create(new RedisOptions()
+            {
+                Factory = t => new TestConnectionMultiplexer(server)
+            });
+            var resolver = new DefaultHubProtocolResolver(new IHubProtocol[]
+            {
+                new JsonHubProtocol(),
+                new MessagePackHubProtocol(),
+            }, NullLogger<DefaultHubProtocolResolver>.Instance);
+
+            _manager1 = new RedisHubLifetimeManager<TestHub>(logger, options, resolver);
+            _manager2 = new RedisHubLifetimeManager<TestHub>(logger, options, resolver);
+
+            async Task ConnectClient(TestClient client, IHubProtocol protocol, string userId, string group)
+            {
+                await _manager2.OnConnectedAsync(HubConnectionContextUtils.Create(client.Connection, protocol, userId));
+                await _manager2.AddGroupAsync(client.Connection.ConnectionId, "Everyone");
+                await _manager2.AddGroupAsync(client.Connection.ConnectionId, group);
+            }
+
+            // Connect clients
+            _clients = new TestClient[ClientCount];
+            var tasks = new Task[ClientCount];
+            for (var i = 0; i < _clients.Length; i++)
+            {
+                var protocol = (i % 2 == 0) ? (IHubProtocol)new JsonHubProtocol() : new MessagePackHubProtocol();
```

Comment:
Have a param for the number of protocol's to serialize? Could just wrap JsonHubProtocol in a test protocol implementation and vary the name slightly to simulate.